pkg/sbom/spdx/unmarshal.go

```diff
@@ -135,11 +142,53 @@ func (s *SPDX) unmarshal(spdxDocument *spdx.Document) error {
 		s.SBOM.Applications = append(s.SBOM.Applications, *app)
 	}
 
+	// Fallback for when there are no effective relationships.
+	if len(relationships) == 0 {
+		if err := s.parsePackages(spdxDocument); err != nil {
+			return err
+		}
+	}
```

Comment:
Can you add testcase for this?

Reply:
Done
https://github.com/aquasecurity/trivy/pull/4210/commits/f6934d53a31937dbcc15b11079b3710c50f23863